task/get-rpm-sources.yaml

```diff
@@ -97,47 +97,121 @@ spec:
+            unknown = " ".join([x for x in values if x.startswith("%")])
+            if unknown:
+                log.warning("Unknown macros in %s: %s", (name_map[name], unknown))
+                return []
+            return values    
+        
+        def get_specfile():
+            specfile_path = glob.glob(os.path.join('/var/workdir/source', '*.spec'))
+            
+            if len(specfile_path) == 0:
+                raise RuntimeError("no spec file available")
+            
+            if len(specfile_path) > 1:
+                raise RuntimeError("too many specfiles: {0}".format(
+                    ', '.join(specfile_path)
+                ))
+            
+            try:
+                spec = Specfile(specfile_path[0])
+            except TypeError as ex:
+                raise RuntimeError("No .spec file") from ex
+            except OSError as ex:
+                raise RuntimeError(ex) from ex
+        
+            return spec
+        
         selected_architectures = sys.argv[1:]
         print(f"Trying to build for {selected_architectures}")
-
+        
+        spec = get_specfile()
+        
+        tags = spec.tags(spec.parsed_sections.package).content
+        arches = {}
+        for name in ['exclusivearch', 'excludearch', 'buildarch']:
+            arches[name] = get_arches(name, tags)
+        
         architecture_decision = {
-          "deps-x86_64": "linux/amd64",
-          "deps-i686": "linux/amd64",
-          "deps-aarch64": "linux/arm64",
-          "deps-s390x": "linux/s390x",
-          "deps-ppc64le": "linux/ppc64le",
-          "build-x86_64": "linux/amd64",
-          "build-i686": "linux/amd64",
-          "build-aarch64": "linux/arm64",
-          "build-s390x": "linux/s390x",
-          "build-ppc64le": "linux/ppc64le",
+            "deps-x86_64": "linux/amd64",
+            "deps-i686": "linux/amd64",
+            "deps-aarch64": "linux/arm64",
+            "deps-s390x": "linux/s390x",
+            "deps-ppc64le": "linux/ppc64le",
+            "build-x86_64": "linux/amd64",
+            "build-i686": "linux/amd64",
+            "build-aarch64": "linux/arm64",
+            "build-s390x": "linux/s390x",
+            "build-ppc64le": "linux/ppc64le",
         }
+        
+        hermetic = 'true'
         # Set the value to 'localhost' if you want to skip the corresponding
         # task (the tasks are modified so they do nothing on localhost).
         if "$(params.hermetic)" != "true":
-          for key in architecture_decision.keys():
-            if key.startswith("deps-"):
-              print(f"non-hermetic build, disabling {key} task")
-              architecture_decision[key] = "localhost"
-
+            for key in architecture_decision.keys():
+                if key.startswith("deps-"):
+                    print(f"non-hermetic build, disabling {key} task")
+                    architecture_decision[key] = "localhost"
+        else:
```

Comment:
This should be done both for {non-}hermetic cases, ie this shouldn't be done in `else` branch.